salt/modules/junos.py

```diff
@@ -794,33 +801,25 @@ def install_config(path=None, **kwargs):
         write_diff = op['diffs_file']
         del op['diffs_file']
 
-    if 'template_path' in op:
-        if not os.path.isfile(op['template_path']):
-            ret['message'] = 'Invlaid template path.'
-            ret['out'] = False
-            return ret
-        if 'template_vars' not in op:
-            ret[
-                'message'] = 'Please provide jinja variable along with the \
-                jina template.'
-            ret['out'] = False
-            return ret
-        if not os.path.isfile(op['template_vars']):
-            ret['message'] = 'Invlaid template_vars path.'
-            ret['out'] = False
-            return ret
-        data = yaml.load(open(op['template_vars']))
-        op['template_vars'] = data
-    else:
-        op = {'path': path}
+    op['path'] = template_cached_path
+
+    if 'format' not in op:
```

Comment:
the issue is that salt will render the template into a temp file named something like /tmp/asdfadf, so we can't rely on file extension after the render operation. I added the code there to catch that. I.e. in case of xml format which isn't caught by pyez